pandas/tests/reshape/concat/test_datetimes.py

```diff
@@ -518,3 +512,15 @@ def test_concat_period_other_series(self):
         result = concat([x, y], ignore_index=True)
         tm.assert_series_equal(result, expected)
         assert result.dtype == "object"
+
+
+def test_concat_timedelta64_block():
+    from pandas import to_timedelta
+
+    rng = to_timedelta(np.arange(10), unit="s")
+
+    df = DataFrame({"time": rng})
+
+    result = concat([df, df])
+    assert (result.iloc[:10]["time"] == rng).all()
```

Comment:
ca n you this use tm.assert_frame_equal instead (and use a fully constructed dataframe)